apis/Google.Cloud.Storage.V1/Google.Cloud.Storage.V1/StorageClientImpl.UpdateObject.cs

```diff
@@ -42,6 +42,8 @@ private ObjectsResource.UpdateRequest CreateUpdateObjectRequest(Object obj, Upda
             GaxPreconditions.CheckArgument(obj.Name != null, nameof(obj), "The Name property of the object to update is null");
             var request = Service.Objects.Update(obj, obj.Bucket, obj.Name);
             options?.ModifyRequest(request, obj);
+            RetryOptions retryOptions = options?.RetryOptions ?? RetryOptions.MaybeIdempotent(options?.IfMetagenerationMatch);
+            RetryHandler.MarkAsRetriable(request, retryOptions);
```

Comment:
Same as PatchObject. I am not sure so want to be doubly sure.

Reply:
The "MarkAsRetriable" statement was previously inside the "ModifyRequest" and just moving it below and all tests are running fine. Also the ApplyEncryptionKey code doesnt seem to be affected by order .

Reply:
Ok. Same as above I wanted to be sure as I am not aware.